### DISCUSSION TITLE
@base selector has higher priority than CSS class.

### DISCUSSION THREAD

**Discussion by codemile** [tailwindlabs/tailwindcss#6977] [Help] [answered]
I'm trying to resolve a problem where the base  stylings from Tailwind for a button has a higher priority than the styles for my React Material UI buttons. Which basically means, all my buttons have a "transparent" background instead of the Material UI theme color. As a result, this is causing the white text of the button to disappear against the white page background under the button.

![image](https://user-images.githubusercontent.com/50146659/148654390-22662c54-d8a2-4572-a5e1-b3b5dc2f7efe.png)

I'm running a NextJs with React Material UI and Tailwind. I feel this is a style order problem in the `<head>` but I've been unable to figure it out.

I could use help resolving this specific problem or help setting up a NextJs+MUI+Tailwind project correctly.

Any help would be appreciated.

**Comment by codemile** [comment, marked as answer]
I had followed the Material UI example project for configuring NextJs. This one found here:

https://github.com/mui-org/material-ui/tree/88aa00a1665d1c6bb642392d104c9188ee830799/examples/nextjs

That example tells you to setup Emotion cache to "prepend" injections into the `<head>`. In addition to that, I had also set the used `<StyledEngineProvider injectFirst={true}>` to inject the dynamic styles into the top of the head.

**So DO NOT do the above.** I fixed by problem by setting prepend to `false` and removing the `StyledEngineProvider` from the code. This seems to have fixed the problem.

I also had my Tailwind configured with an important selector. I set a `important: '#app'` in my `tailwind.config.js` file. I then set the `<html lang="en" id="app">` tag with the ID. This is done because the "dark" class is applied to the `<body>` element, and you can't set both the dark class and important selector on the same element.

I hope this helps someone in the future.

**Reply by liho00** [reply]
> I had followed the Material UI example project for configuring NextJs. This one found here:
> 
> https://github.com/mui-org/material-ui/tree/88aa00a1665d1c6bb642392d104c9188ee830799/examples/nextjs
> 
> That example tells you to setup Emotion cache to "prepend" injections into the `<head>`. In addition to that, I had also set the used `<StyledEngineProvider injectFirst={true}>` to inject the dynamic styles into the top of the head.
> 
> **So DO NOT do the above.** I fixed by problem by setting prepend to `false` and removing the `StyledEngineProvider` from the code. This seems to have fixed the problem.
> 
> I also had my Tailwind configured with an important selector. I set a `important: '#app'` in my `tailwind.config.js` file. I then set the `<html lang="en" id="app">` tag with the ID. This is done because the "dark" class is applied to the `<body>` element, and you can't set both the dark class and important selector on the same element.
> 
> I hope this helps someone in the future.

Thanks bro you saved my day. Love u deep deep <3

**Comment by Elly-Zhou** [comment]
Refer to: [stackoverflow: tailwind is not working](https://stackoverflow.com/questions/77200886/tailwind-is-not-working-together-with-styled-material-ui-components-in-nextjs-13)
Watch out the step 2 & 3 in [this link](https://mui.com/material-ui/integrations/interoperability/#tailwind-css), that would be work.
